### PR TITLE
Add Github actions

### PR DESCRIPTION
**Pull request by Majed6**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - run: sudo docker run --name mongo -d -p 27017:27017 mongo
+      - name: Checkout Habitica
+        uses: actions/checkout@v2
+        with:
+          path: habitica
+          repository: HabitRPG/habitica
+          ref: 'develop'
```
Comment (paglias) [inline .github/workflows/test.yml]: could this be changed to `release`?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      - run: |
+          cd habitica
+          cp config.json.example config.json
+          npm ci
+          node ./website/server/index.js & cd ..
```
Comment (paglias) [inline .github/workflows/test.yml:18]: `npm start` might be better here ?

**Review by paglias — COMMENTED** [review]
thanks a lot @Majed6 ! This is looking good, I just left a couple of comments. I'll try to merge this as soon as possible so we can see how it's working!

File: .github/workflows/test.yml (hunk 1 above)
Comment (Majed6) [reply .github/workflows/test.yml]: Done

File: .github/workflows/test.yml (hunk 2 above)
Comment (Majed6) [reply .github/workflows/test.yml:18]: `npm start` runs `gulp nodemon` . I couldn't get it running on github pipeline.

```
> gulp nodemon

[05:16:33] Using gulpfile ~/work/habitica-zapier/habitica-zapier/habitica/gulpfile.js
[05:16:33] Starting 'nodemon'...
[05:16:33] Finished 'nodemon' after 30 ms
[05:16:33] [nodemon] 1.19.4
[05:16:33] [nodemon] to restart at any time, enter `rs`
[05:16:33] [nodemon] watching dir(s): *.*
[05:16:33] [nodemon] watching extensions: js,mjs,json
[05:16:33] [nodemon] starting `node ./website/server/index.js`
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! habitica@4.136.0 start: `gulp nodemon`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the habitica@4.136.0 start script.
```

Comment (paglias) [reply .github/workflows/test.yml:18]: makes sense, probably `nodemon` doesn't run on GH actions

**Comment by paglias** [comment]
first run went well! thanks @Majed6 , noted towards your 3rd tier

**Pull request merged**